src/commands/file.ts

```diff
@@ -212,16 +210,17 @@ export function registerFileCommands(context: ExtensionContext) {
           openLabel: "Upload"
         });
 
+
         if (files) {
           withProgress("Uploading file(s)...", () =>
             Promise.all(
-              files.map((file) => {
+              files.map(async (file) => {
                 const fileName = path.basename(file.path);
-                const content = fs.readFileSync(new URL(file.toString()));
+                const content = workspace.fs.readFile(file);
 
                 return workspace.fs.writeFile(
                   fileNameToUri(node.gist.id, fileName),
-                  content
+                  await content
```

Comment:
Could we move the await to line 219? That way it's clear that "content" refers to the actual content and not a promise?

Reply:
Yes.  Moved, rebuilt .vsix, re-tested.
https://github.com/lostintangent/gistpad/pull/299/commits/4af9790d7858b0c8f5e0787e15ad143a7a8e6806